public/app/features/alerting/unified/components/settings/VersionManager.tsx

```diff
@@ -0,0 +1,314 @@
+                  }}
+                >
+                  Compare
+                </Button>
+                <Button
+                  variant="secondary"
+                  size="sm"
+                  icon="history"
+                  onClick={() => {
+                    setActiveRestoreVersion(versionID);
+                    showConfirmation();
+                  }}
+                  disabled={restoreVersionState.isLoading}
+                >
+                  Restore
+                </Button>
+              </>
+            )}
+          </Stack>
+        );
+      },
+    },
+  ];
+
+  if (restoreVersionState.isLoading) {
+    return (
+      <Alert severity="info" title="Restoring Alertmanager configuration">
+        This might take a while...
+      </Alert>
+    );
+  }
+
+  return (
+    <>
+      {activeComparison ? (
+        <CompareVersions
+          left={activeComparison[0]}
+          right={activeComparison[1]}
+          disabled={restoreVersionState.isLoading}
+          onCancel={() => {
+            setActiveRestoreVersion(undefined);
+            setActiveComparison(undefined);
+            hideConfirmation();
+          }}
+          onConfirm={() => {
+            showConfirmation();
+          }}
+        />
+      ) : (
+        <InteractiveTable pageSize={VERSIONS_PAGE_SIZE} columns={columns} data={rows} getRowId={(row) => row.id} />
+      )}
+      {/* TODO make this modal persist while restore is in progress */}
+      <ConfirmModal
+        isOpen={confirmRestore}
+        title={'Restore Version'}
+        body={'Are you sure you want to restore the configuration to this version? All unsaved changes will be lost.'}
+        confirmText={'Yes, restore configuration'}
+        onConfirm={() => {
+          if (activeRestoreVersion) {
+            restoreVersion(activeRestoreVersion);
+          }
+
+          hideConfirmation();
+        }}
+        onDismiss={() => hideConfirmation()}
+      />
+    </>
+  );
+};
+
+interface CompareVersionsProps {
+  left: string;
+  right: string;
+
+  disabled?: boolean;
+  onCancel: () => void;
+  onConfirm: () => void;
+}
+
+function CompareVersions({ left, right, disabled = false, onCancel, onConfirm }: CompareVersionsProps) {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div className={styles.drawerWrapper}>
+      <div className={styles.diffWrapper}>
+        <DiffViewer newValue={left} oldValue={right} />
+      </div>
+      <Stack direction="row" alignItems="center">
+        <Spacer />
+        <Button variant="secondary" onClick={onCancel} disabled={disabled}>
+          Return
+        </Button>
+        <Button icon="history" variant="primary" onClick={onConfirm} disabled={disabled}>
+          Restore
+        </Button>
+      </Stack>
+    </div>
+  );
+}
+
+const LastAppliedCell = ({ value }: CellProps<VersionData>) => {
+  const date = moment(value);
```

Comment:
Should we try avoid using moment and use `date-fns` instead? :)

Reply:
Hm I think I used `moment` here because I could re-use the same `DateTime` object for both relative and absolute timestamps here but I've no strong opinion on which one we prefer :D